### PR TITLE
refactor(plugin/digitization): Add Digitization-specific SourceLink (EDM rework part 2a/4)

### PR DESCRIPTION
**Pull request by msmk0**
Remove the dependency of the digitization plugin on the generic `Identifier` and the `IdentificationPlugin`. Introduce a custom `DigitizationSourceLink` instead, that fulfills the `SourceLinkConcept` and contains all the necessary information.

### CODE REVIEW

**Comment by asalzburger** [comment]
As part of the #497 the `DigitizationPlugin` will be obsolete (as will be the `IdentificationPlugin`), hence this PR will make it easier to do that shift.

**Comment by msmk0** [comment]
@asalzburger Could you review this?

File: Plugins/Digitization/include/Acts/Plugins/Digitization/DigitizationSourceLink.hpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  friend bool operator==(const DigitizationSourceLink& lhs,
+                         const DigitizationSourceLink& rhs) {
+    return (lhs.m_surface == rhs.m_surface) and
+           (lhs.m_indices == rhs.m_indices);
+  }
+  friend bool operator!=(const DigitizationSourceLink& lhs,
+                         const DigitizationSourceLink& rhs) {
+    return not(lhs == rhs);
+  }
```
Comment (paulgessinger) [inline Plugins/Digitization/include/Acts/Plugins/Digitization/DigitizationSourceLink.hpp:61]: Sorry, I can't help myself.

[Use a member when you can, and a friend when you have to.](https://isocpp.org/wiki/faq/friends#member-vs-friend-fns)

Comment (msmk0) [reply Plugins/Digitization/include/Acts/Plugins/Digitization/DigitizationSourceLink.hpp:61]: 😄 I will change it.

Comment (msmk0) [reply Plugins/Digitization/include/Acts/Plugins/Digitization/DigitizationSourceLink.hpp]: Fixed.

File: Plugins/Digitization/include/Acts/Plugins/Digitization/DigitizationSourceLink.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  /// satisfy SourceLinkConcept.
+  DigitizationSourceLink() = default;
+  DigitizationSourceLink(const DigitizationSourceLink&) = default;
+  DigitizationSourceLink(DigitizationSourceLink&&) = default;
+  DigitizationSourceLink& operator=(const DigitizationSourceLink&) = default;
+  DigitizationSourceLink& operator=(DigitizationSourceLink&&) = default;
+
+  /// Access the reference surface.
+  const Surface& referenceSurface() const {
+    assert(m_surface and "Invalid Surface pointer in DigitizationSourceLink");
```
Comment (asalzburger) [inline Plugins/Digitization/include/Acts/Plugins/Digitization/DigitizationSourceLink.hpp:41]: I see, with this trick you get the assert message printed out, right?
We haven'd done this like this in other areas, we should streamline one method.

**Review by asalzburger — APPROVED** [review]
That's straight forward, and as mentioned, this will somewhat become obsolete in a few days ...

**Pull request merged**